diff --git a/plugins/de.cau.cs.kieler.kiml.graphviz.dot/.project b/plugins/de.cau.cs.kieler.kiml.graphviz.dot/.project
index c3379c3..b502453 100644
--- a/plugins/de.cau.cs.kieler.kiml.graphviz.dot/.project
+++ b/plugins/de.cau.cs.kieler.kiml.graphviz.dot/.project
@@ -6,6 +6,11 @@
 	</projects>
 	<buildSpec>
 		<buildCommand>
+			<name>org.eclipse.xtext.ui.shared.xtextBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
 			<name>org.eclipse.jdt.core.javabuilder</name>
 			<arguments>
 			</arguments>
@@ -34,7 +39,7 @@
 	<natures>
 		<nature>org.eclipse.jdt.core.javanature</nature>
 		<nature>org.eclipse.pde.PluginNature</nature>
-		<nature>org.eclipse.xtext.ui.shared.xtextNature</nature>
 		<nature>net.sf.eclipsecs.core.CheckstyleNature</nature>
+		<nature>org.eclipse.xtext.ui.shared.xtextNature</nature>
 	</natures>
 </projectDescription>
diff --git a/plugins/de.cau.cs.kieler.kiml.graphviz.layouter/src/de/cau/cs/kieler/kiml/graphviz/layouter/GraphvizTool.java b/plugins/de.cau.cs.kieler.kiml.graphviz.layouter/src/de/cau/cs/kieler/kiml/graphviz/layouter/GraphvizTool.java
index 14c7a32..028065a 100644
--- a/plugins/de.cau.cs.kieler.kiml.graphviz.layouter/src/de/cau/cs/kieler/kiml/graphviz/layouter/GraphvizTool.java
+++ b/plugins/de.cau.cs.kieler.kiml.graphviz.layouter/src/de/cau/cs/kieler/kiml/graphviz/layouter/GraphvizTool.java
@@ -103,13 +103,13 @@
             }
         }
         
-        if (DEFAULT_LOCS.isEmpty()) {
+//        if (DEFAULT_LOCS.isEmpty()) {
             // Fallback list of default locations
             DEFAULT_LOCS.add("/opt/local/bin/");
             DEFAULT_LOCS.add("/usr/local/bin/");
             DEFAULT_LOCS.add("/usr/bin/");
             DEFAULT_LOCS.add("/bin/");
-        }
+//        }
     }
     
     
@@ -194,19 +194,20 @@
      */
     public static String getDotExecutable(final boolean promptUser) {
         // load the graphviz path from the preferences, if any
-        IPreferenceStore preferenceStore = GraphvizLayouterPlugin.getDefault().getPreferenceStore();
-        String dotExecutable = preferenceStore.getString(PREF_GRAPHVIZ_EXECUTABLE);
-        File dotFile = new File(dotExecutable);
-        
-        // check if the executable exists
-        if (!dotFile.exists() || !dotFile.canExecute()) {
+//        IPreferenceStore preferenceStore = GraphvizLayouterPlugin.getDefault().getPreferenceStore();
+//        String dotExecutable = preferenceStore.getString(PREF_GRAPHVIZ_EXECUTABLE);
+//        File dotFile = new File(dotExecutable);
+//        
+//        // check if the executable exists
+//        if (!dotFile.exists() || !dotFile.canExecute()) {
+    		String dotExecutable = null;
             boolean foundExec = false;
             
             // look in a selection of default locations where it might be installed
             for (String location : DEFAULT_LOCS) {
                 // Linux
                 dotExecutable = location + "dot";
-                dotFile = new File(dotExecutable);
+                File dotFile = new File(dotExecutable);
                 if (dotFile.exists() && dotFile.canExecute()) {
                     foundExec = true;
                     break;
@@ -222,15 +223,15 @@
             }
             
             if (!foundExec) {
-                if (promptUser) {
-                    handleExecPath();
-                    // fetch the executable string again after the user has entered a new path
-                    dotExecutable = preferenceStore.getString(PREF_GRAPHVIZ_EXECUTABLE);
-                } else {
+//                if (promptUser) {
+//                    handleExecPath();
+//                    // fetch the executable string again after the user has entered a new path
+//                    dotExecutable = preferenceStore.getString(PREF_GRAPHVIZ_EXECUTABLE);
+//                } else {
                     return null;
                 }
-            }
-        }
+//            }
+//        }
         
         return dotExecutable;
     }
@@ -500,13 +501,13 @@
                     }
                 }
                 
-                // retrieve the current timeout value
-                IPreferenceStore preferenceStore =
-                        GraphvizLayouterPlugin.getDefault().getPreferenceStore();
-                int timeout = preferenceStore.getInt(PREF_TIMEOUT);
-                if (timeout < PROCESS_MIN_TIMEOUT) {
-                    timeout = PROCESS_DEF_TIMEOUT;
-                }
+//                // retrieve the current timeout value
+//                IPreferenceStore preferenceStore =
+//                        GraphvizLayouterPlugin.getDefault().getPreferenceStore();
+//                int timeout = preferenceStore.getInt(PREF_TIMEOUT);
+//                if (timeout < PROCESS_MIN_TIMEOUT) {
+                   int timeout = PROCESS_DEF_TIMEOUT;
+//                }
                 
                 boolean interrupted = false;
                 try {